### PR TITLE
Plumb slot update pubsub notifications

### PR DESCRIPTION
**Pull request by carllin**
#### Summary of Changes

Fixes #

### CODE REVIEW

**Review by jstarry — DISMISSED** [review]
looks good once ci is fixed

File: client/src/rpc_response.rs [hunk 1]
```diff
@@ -101,6 +101,16 @@ pub struct SlotInfo {
     pub root: Slot,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", tag = "type")]
+pub enum SlotUpdate {
+    OptimisticConfirmation { slot: Slot, timestamp: u64 },
```
Comment (sakridge) [inline client/src/rpc_response.rs:107]: How about errored blocks? Missing I guess we know if there is just no update for a given slot.

Comment (carllin) [reply client/src/rpc_response.rs:107]: hmmm yeah if we sent all the roots, then you could tell which ones are missing due to wrong fork/errors. If we want to distinguish between wrong fork/replayed error we could add an extra errored block notification

File: core/src/rpc_subscriptions.rs [hunk 2]
```diff
@@ -387,6 +393,8 @@ struct Subscriptions {
     gossip_program_subscriptions: Arc<RpcProgramSubscriptions>,
     gossip_signature_subscriptions: Arc<RpcSignatureSubscriptions>,
     slot_subscriptions: Arc<RpcSlotSubscriptions>,
+    // TODO: Figure out when to remove a subscriber (some number of slots)
```
Comment (jstarry) [inline core/src/rpc_subscriptions.rs:396]: this can be removed now :)

**Review by jstarry — APPROVED** [review]

File: core/src/retransmit_stage.rs [hunk 3]
```diff
@@ -341,6 +382,16 @@ fn retransmit(
...
             max_slot = max_slot.max(shred_slot);
+
+            if let Some(rpc_subscriptions) = rpc_subscriptions {
+                notify_first_shred_received(
+                    shred_slot,
+                    rpc_subscriptions,
+                    sent_received_slot_notification,
+                    &root_bank,
+                );
+            }
```
Comment (behzadnouri) [inline core/src/retransmit_stage.rs:393]: @carllin Would it be ok if we move this rpc notification to window-service? 
Or is there some reason this notification is sent in retransmit-stage?

**Pull request merged**